src/plugins/vis_type_timeline/server/routes/run.ts

```diff
@@ -106,7 +105,8 @@ export function runRoute(
           savedObjectsClient: context.core.savedObjects.client,
         });
         const chainRunner = chainRunnerFn(tlConfig);
-        const sheet = await Bluebird.all(chainRunner.processRequest(request.body));
+        const promises = chainRunner.processRequest(request.body);
+        const sheet = await Promise.all(Array.from(promises));
```

Comment:
Does `chainRunner.processRequest(request.body)` returns returns an array of promises? or it returns an iterable (like a Set) of promises, so you convert it to an array before passing it to Promise.all?

From `chainRunnerFn` def, I see the return is
```
return {
    processRequest: processRequest,
    getStats: function () {
      return stats;
    },
  };
```
where `processRequest` is defined as
```
function processRequest(request) {
    if (!request) throw new Error('Empty request body');

    validateTime(request.time, tlConfig);

    tlConfig.time = request.time;
    tlConfig.time.to = moment(request.time.to).valueOf();
    tlConfig.time.from = moment(request.time.from).valueOf();
    tlConfig.time.interval = calculateInterval(
      tlConfig.time.from,
      tlConfig.time.to,
      tlConfig.settings['timeline:target_buckets'] || 200,
      tlConfig.time.interval,
      tlConfig.settings['timeline:min_interval'] || '1ms'
    );

    tlConfig.setTargetSeries();

    stats.invokeTime = new Date().getTime();
    stats.queryCount = 0;
    queryCache = {};

    // This is setting the "global" sheet, required for resolving references
    sheet = parseSheet(request.sheet);
    return preProcessSheet(sheet).then(function () {
      return _.map(sheet, function (chainList, i) {
        return resolveChainList(chainList)
          .then(function (seriesList) {
            stats.sheetTime = new Date().getTime();
            return seriesList;
          })
          .catch(function (e) {
            throwWithCell(i, e);
          });
      });
    });
  }
```
The `_.map` function from lodash is being used here.`resolveChainList(chainList)` returns a Promise. ` _.map` returns an array. So it seems to me the return type is array of Promise. Therefore, we could do
```
const sheet = await Promise.all(chainRunner.processRequest(request.body));
```
? 

Could you help me to double check and verify? Could set some console logs to verify the type of promises chainRunner returns. 

Since this is in server, you could add launch.json in .vscode then add a break point when you run a timeline func
```
{
    "version": "0.2.0",
    "configurations": [
        {
            "type": "pwa-node",
            "request": "attach",
            "name": "back",
            "port": 9229
        }
    ]
}
```